sphinx/writers/latex.py

```diff
@@ -958,7 +958,14 @@ def depart_seealso(self, node: Element) -> None:
     def visit_rubric(self, node: Element) -> None:
         if len(node) == 1 and node.astext() in ('Footnotes', _('Footnotes')):
             raise nodes.SkipNode
-        self.body.append(r'\subsubsection*{')
+        tag = {
+            1: 'section',
+            2: 'subsection',
+            3: 'subsubsection',
+            4: 'paragraph',
+            5: 'subparagraph',
+        }.get(node.get('level', 0), 'subsubsection')
```

Comment:
Add also an assert here because people are not meant to inject a level deeper than what should be allowed. And possibly add a warning to indicate that a level outside 0-5 (and not 0-6!) is not allowed.

Actually, if someone uses level 6, then it's fine in HTML but in LaTeX you would have a subsubsection which may bigger than a subpargraph. So for level 6, it should 'subparagraph' (I'm not sure if the document class has a subparagraph, maybe you should check all levels + non-levels in tests)

Reply:
This now uses the "correct" logic, using the `self.sectionnames` and `self.top_sectionlevel` variables.

a warning for unsupported level values is also added